src/parser.cpp

```diff
@@ -1125,4 +1157,216 @@ void Parser::ParserImpl::loadImport(const ImportSourcePtr &importSource, const M
+                                    "' referencing variable '" + variableName +
+                                    "' has an invalid non-whitespace child text element '" + textNode + "'.");
+                err->setReset(reset);
+                err->setKind(Error::Kind::RESET);
+                mParser->addError(err);
+            }
+        } else {
+            ErrorPtr err = std::make_shared<Error>();
+            err->setDescription("Reset in component '" + component->getName() +
+                                "' referencing variable '" + variableName +
+                                "' has an invalid child element '" + childNode->getType() + "'.");
+            err->setReset(reset);
+            err->setKind(Error::Kind::RESET);
+            mParser->addError(err);
+        }
+        childNode = childNode->getNext();
+    }
+}
+
+void Parser::ParserImpl::loadWhen(const WhenPtr &when, const ResetPtr &reset, const XmlNodePtr &node)
+{
+    std::string referencedVariableName = "";
+    VariablePtr referencedVariable =  reset->getVariable();
+    if (referencedVariable != nullptr) {
+        referencedVariableName = referencedVariable->getName();
+    }
+    std::string resetOrder = "";
+    if (reset->isOrderSet()) {
+        resetOrder = convertIntToString(reset->getOrder());
+    }
+    int order = 0;
+    bool orderDefined = false;
+    XmlAttributePtr attribute = node->getFirstAttribute();
+    while (attribute) {
+        if (attribute->isType("order")) {
+            if (convertToInt(attribute->getValue(), &order)) {
+                orderDefined = true;
+            }
+        } else if (attribute->isType("id")) {
+            when->setId(attribute->getValue());
+        } else {
+            ErrorPtr err = std::make_shared<Error>();
+            err->setDescription("When in reset referencing variable '" + referencedVariableName +
+                                "' with order '" + resetOrder +
+                                "' has an invalid attribute '" + attribute->getType() + "'.");
+            err->setWhen(when);
+            err->setKind(Error::Kind::WHEN);
+            mParser->addError(err);
+        }
+        attribute = attribute->getNext();
+    }
+
+    if (orderDefined) {
+        when->setOrder(order);
+    } else {
+        ErrorPtr err = std::make_shared<Error>();
+        err->setDescription("When in reset referencing variable '" + referencedVariableName +
+                            "' with order '" + resetOrder +
+                            "' does not have an order defined.");
+        err->setWhen(when);
+        err->setKind(Error::Kind::WHEN);
+        mParser->addError(err);
+    }
+
+    size_t mathNodeCount = 0;
+    XmlNodePtr childNode = node->getFirstChild();
+    while (childNode) {
+        if (childNode->isType("math")) {
+            // TODO: copy any namespaces declared in parents into the math element
+            //       so math is a valid subdocument.
```

Comment:
Shouldn't this be done before having this PR merged in?

Reply:
I'm not finding any problems with this at this point so I don't have any desire to fix something that is not broken.  If we have an example that we can test against that does cause a problem then I would fix it.

Reply:
Fair enough, it's just that whenever I see a `TODO`, then I would expect it implemented or deleted (i.e. have the code as clean as possible).